### PR TITLE
Move Tracers to top-level dependency position.

### PR DESCRIPTION
**Pull request by dnoe**
Thanks to a new static registration process optional tracers like
Lightstep and Zipkin can be added as top level dependencies of the site
specific main build.  This change cleans up the remaining items required
to make a site specific main that does not link in Lightstep and Zipkin
libraries.

### CODE REVIEW

**Comment by dnoe** [comment]
@mrice32

**Comment by mrice32** [comment]
LGTM

File: source/common/tracing/BUILD [hunk 1]
```diff
@@ -39,3 +36,17 @@ envoy_cc_library(
         "//source/common/runtime:uuid_util_lib",
     ],
 )
+
+envoy_cc_library(
+    name = "lightstep_tracer_lib",
+    srcs = [
```
Comment (htuch) [inline source/common/tracing/BUILD]: Nit: Very slight preference for `srcs = ["foobar.cc"],` style when there is only one item in the list.

Comment (dnoe) [inline source/common/tracing/BUILD]: If it's OK with you I'd rather not wait for another 40 minute travis build for this.  I'll do single line unless buildifier overrules in the future.

**Pull request merged**